dhctl/pkg/config/meta.go

```diff
@@ -134,6 +138,97 @@ func (m *MetaConfig) Prepare(ctx context.Context, preparatorProvider MetaConfigP
 	return validateAndPrepareMetaConfig(ctx, preparatorProvider, m)
 }
 
+func (m *MetaConfig) prepareRegistry() error {
+	var (
+		defaultCRI        string
+		deckhouseSettings *registry_moduleconfig.DeckhouseSettings
+		initConfig        *initconfig.Config
+	)
+
+	// Extract defaultCRI
+	if rawCRI, exists := m.ClusterConfig["defaultCRI"]; exists {
+		if err := json.Unmarshal(rawCRI, &defaultCRI); err != nil {
+			return fmt.Errorf("get defaultCRI from cluster config: %w", err)
+		}
+	}
+	criSupported := registry_const.IsCRISupported(defaultCRI)
+
+	// Extract configuration from initConfig
+	if m.DeckhouseConfig.ImagesRepo != "" ||
+		m.DeckhouseConfig.RegistryDockerCfg != "" ||
+		m.DeckhouseConfig.RegistryCA != "" {
```

Comment:
it looks like this comparsion chain should be wrapped to the method in `DeckhouseConfig`